[Turbo] Document `<twig:Turbo:Stream:*>` (#2481)
symfony/ux: src/Turbo/doc/index.rst

    {# bottom of new.html.twig #}
    {% block success_stream %}
        <turbo-stream action="replace" targets="#my_div_id">
            <template>
                The element having the id "my_div_id" will be replaced by this block, without a full page reload!

                <div>The task "{{ task }}" has been created!</div>
            </template>
        </turbo-stream>
    {% endblock %}

Supported actions are ``append``, ``prepend``, ``replace``, ``update``,
``remove``, ``before``, ``after`` and ``refresh``.
`Read the Turbo Streams documentation for more details`_.

Stream Messages and Actions
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> To render a ``<turbo-stream>`` element, we recommend using the ``<twig:Turbo:Stream:*>`` Twig Component to avoid typos.

Reviewer: ```suggestion
To render a ``<turbo-stream>`` element, this bundle provides a set of ``<twig:Turbo:Stream:*>`` Twig Components. These components make it easy to inject content directly into the ``<template>`` tag, pass attributes, and set the desired morphing mode with a clear and consistent syntax.
```

    {# bottom of new.html.twig #}
    {% block success_stream %}
        <turbo-stream action="replace" targets="#my_div_id">
            <template>
                The element having the id "my_div_id" will be replaced by this block, without a full page reload!

                <div>The task "{{ task }}" has been created!</div>
            </template>
        </turbo-stream>
    {% endblock %}

Supported actions are ``append``, ``prepend``, ``replace``, ``update``,
``remove``, ``before``, ``after`` and ``refresh``.
`Read the Turbo Streams documentation for more details`_.

Stream Messages and Actions
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> To render a ``<turbo-stream>`` element, we recommend using the ``<twig:Turbo:Stream:*>`` Twig Component to avoid typos.

Reviewer 2: Is this the only reason we added `Turbo:Stream:*` components?

Reviewer: Fixed / thx

Append
""""""

.. code-block:: html+twig

    <twig:Turbo:Stream:Append target="#dom_id">
        Content to append to container designated with the dom_id.
    </twig:Turbo:Stream:Append>

    {# renders as: #}

>     <turbo-stream action="append" targets="#dom_id">

Reviewer 2: `target` or `targets`?

Reviewer 2: (same comment applies for other examples)

Author: `target` becomes `targets`

        <template>
            Content to append to container designated with the dom_id.
        </template>
    </turbo-stream>

> 

Reviewer: ```suggestion
```

> For more info, see: `Turbo Streams - Append <https://turbo.hotwired.dev/reference/streams#append>`_.

Reviewer: ```suggestion
```

Prepend
"""""""

.. code-block:: html+twig

    <twig:Turbo:Stream:Prepend target="#dom_id">
        Content to prepend to container designated with the dom_id.
    </twig:Turbo:Stream:Prepend>

    {# renders as: #}
    <turbo-stream action="prepend" targets="#dom_id">
        <template>
            Content to prepend to container designated with the dom_id.
        </template>
    </turbo-stream>

> 

Reviewer: ```suggestion
```

> For more info, see: `Turbo Streams - Prepend <https://turbo.hotwired.dev/reference/streams#prepend>`_.

Reviewer: I don't think we need to add these links every paragraph, as they do not bring much more to the table.

But i think we could probably show the effect on the code... and add a matching page/demo on ux.symfony.com to illustrate the effect on the various actions!

Author: Indeed, it doesn't add much more but it was mainly to show the rendering of each ``<twig:Turbo:Stream:*>`` Twig Component.

What do you mean by "show the effect on the code"?
Concerning a new page on ux.symfony.com, should we make a demo for the 8 actions?

Reviewer: Let's remove all these links, I really do not think one link every 8 lines is a good idea.. If we do not have any content to write, let's keep things light here.

Reviewer: And we will see later for the website, it was another topic :)

> For more info, see: `Turbo Streams - Prepend <https://turbo.hotwired.dev/reference/streams#prepend>`_.

Reviewer: ```suggestion
```

Replace
"""""""

.. code-block:: html+twig

    <twig:Turbo:Stream:Replace target="#dom_id">
        Content to replace the element designated with the dom_id.
    </twig:Turbo:Stream:Replace>

    {# renders as: #}
    <turbo-stream action="replace" targets="#dom_id">
        <template>
            Content to replace the element designated with the dom_id.
        </template>
    </turbo-stream>

.. code-block:: html+twig

    {# with morphing #}
    <twig:Turbo:Stream:Replace target="#dom_id" morph>
        Content to replace the element.
    </twig:Turbo:Stream:Replace>

    {# renders as: #}
    <turbo-stream action="replace" targets="#dom_id" method="morph">
        <template>
            Content to replace the element.
        </template>
    </turbo-stream>

> 

Reviewer: ```suggestion
```

> For more info, see: `Turbo Streams - Replace <https://turbo.hotwired.dev/reference/streams#replace>`_.

Reviewer: ```suggestion
```

Update
""""""

.. code-block:: html+twig

    <twig:Turbo:Stream:Update target="#dom_id">
        Content to update to container designated with the dom_id.
    </twig:Turbo:Stream:Update>

    {# renders as: #}
    <turbo-stream action="update" targets="#dom_id">
        <template>
            Content to update to container designated with the dom_id.
        </template>
    </turbo-stream>

.. code-block:: html+twig

    {# with morphing #}
    <twig:Turbo:Stream:Update target="#dom_id" morph>
        Content to replace the element.
    </twig:Turbo:Stream:Update>

    {# renders as: #}
    <turbo-stream action="update" targets="#dom_id" method="morph">
        <template>
            Content to replace the element.
        </template>
    </turbo-stream>

> 

Reviewer: ```suggestion
```

> For more info, see: `Turbo Streams - Update <https://turbo.hotwired.dev/reference/streams#update>`_.

Reviewer: ```suggestion
```

Remove
""""""

.. code-block:: html+twig

    <twig:Turbo:Stream:Remove target="#dom_id" />

    {# renders as: #}
    <turbo-stream action="remove" targets="#dom_id"></turbo-stream>

> 

Reviewer: ```suggestion
```

> For more info, see: `Turbo Streams - Remove <https://turbo.hotwired.dev/reference/streams#remove>`_.

Reviewer: ```suggestion
```

Before
""""""

.. code-block:: html+twig

    <twig:Turbo:Stream:Before target="#dom_id">
        Content to place before the element designated with the dom_id.
    </twig:Turbo:Stream:Before>

    {# renders as: #}
    <turbo-stream action="before" targets="#dom_id">
        <template>
            Content to place before the element designated with the dom_id.
        </template>
    </turbo-stream>

> 

Reviewer: ```suggestion
```

> For more info, see: `Turbo Streams - Before <https://turbo.hotwired.dev/reference/streams#before>`_.

Reviewer: ```suggestion
```

After
"""""

.. code-block:: html+twig

    <twig:Turbo:Stream:Refresh target="#dom_id">
        Content to place after the element designated with the dom_id.
    </twig:Turbo:Stream:After>

    {# renders as: #}
    <turbo-stream action="after" targets="#dom_id">
        <template>
            Content to place after the element designated with the dom_id.
        </template>
    </turbo-stream>

> 

Reviewer: ```suggestion
```

> For more info, see: `Turbo Streams - After <https://turbo.hotwired.dev/reference/streams#after>`_.

Reviewer: ```suggestion
```

Refresh
"""""""

.. code-block:: html+twig

    {# without [request-id] #}
    <twig:Turbo:Stream:Refresh />

    {# renders as: #}
    <turbo-stream action="refresh"></turbo-stream>

.. code-block:: html+twig

    {# debounced with [request-id] #}
    <twig:Turbo:Stream:Refresh requestId="abcd-1234" />

    {# renders as: #}
    <turbo-stream action="refresh" request-id="abcd-1234"></turbo-stream>

> 

Reviewer: ```suggestion
```

> For more info, see: `Turbo Streams - Refresh <https://turbo.hotwired.dev/reference/streams#refresh>`_.

Reviewer: ```suggestion
```
